tests/models/test_arch_vanilla.py

```diff
@@ -52,3 +52,31 @@ def test_functional() -> None:
     # skipcq
     with pytest.raises(ValueError, match=r".*Backbone.*not supported.*"):
         CNNModel("shiny_model_to_crash", num_classes=2)
+
+
+def test_timm_functional() -> None:
+    """Test for creating backbone."""
+    backbones = [
+        "uni_v1",
+        "prov-gigapath",
+    ]
+    assert TimmModel.postproc([1, 2]) == 1
+
+    b = 4
+    h = w = 224
+    samples = torch.from_numpy(RNG.random((b, h, w, 3)))
+
+    # Dummy entry, will generate ValueError if "try" fails without running the loop.
+    backbone = "empty"
+    try:
+        for backbone in backbones:
+            model = TimmModel(backbone=backbone, num_classes=1, pretrained=False)
```

Comment:
There is a discrepancy between my local tests and the GitHub actions tests when I run
`pytest -svvv tests/models/test_arch_vanilla.py`

I introduced `pretrained` flag in [deea74f](https://github.com/TissueImageAnalytics/tiatoolbox/pull/856/commits/deea74f0e5f9549590a736617c9f9dc5e76f125c)

I renamed the file with my HF token to mimic the situation on GitHub.

If I use `pretrained=True`, I get an error locally
```
E               huggingface_hub.utils._errors.GatedRepoError: 401 Client Error. (Request ID: Root=1-6710df61-738432cb51e75c6b39dd6329;8eb03fe3-9536-4777-8295-a54990368d2e)
E               
E               Cannot access gated repo for url https://huggingface.co/MahmoodLab/UNI/resolve/main/model.safetensors.
E               Access to model MahmoodLab/UNI is restricted. You must have access to it and be authenticated to access it. Please log in.
```

But if I use the `pretrained=False` locally, all tests pass.

On GitHub, even using `pretrained=False` results in a slightly different error. It is no longer about weights but the configuration json file.
```
E               huggingface_hub.errors.GatedRepoError: 401 Client Error. (Request ID: Root=1-6710d846-63696e761354213311810b36;300f8cc2-1e6b-405a-8c99-ebc4470c6615)
E               
E               Cannot access gated repo for url https://huggingface.co/MahmoodLab/UNI/resolve/main/config.json.
E               Access to model MahmoodLab/UNI is restricted. You must have access to it and be authenticated to access it. Please log in.
```

Reply:
The reason we need tests is to make sure that we update the code to any changes in API in future. Can we find a timm model which does not require login? We do not need to test each and every model on GitHub actions. If we can verify the API works that should be fine.

Reply:
I’ll add the smallest EfficientNet

`model = timm.create_model('efficientnet_b0', pretrained=True)`